scipy/special/_basic.py

```diff
@@ -2886,6 +2886,53 @@ def _exact_factorialx_array(n, k=1):
     return out
 
 
+def _factorialx_approx_core(n, k):
+    """
+    Core approximation to multifactorial for array n and integer k.
+    """
+    if k == 1:
+        # shortcut for k=1
+        result = gamma(n + 1)
+        if isinstance(n, np.ndarray):
+            # gamma does not maintain 0-dim arrays
+            result = np.array(result)
+        return result
+
+    n_mod_k = n % k
+    if not isinstance(n, np.ndarray):
```

Comment:
There's a comment directly below?

Reply:
Sorry, what I meant was, when I saw the scalar branch comment I expected to see `np.scalar` but having read the issue maybe that's just an anti-pattern I need to erase from my brain.

Reply:
OK, I get your point now; I'll think about how I document this so the comment isn't duplicated X times.

Reply:
PS. Some of this will change once the complex extensions land; I have a branch that unifies all the underlying implementations of `factorial{,2,k}`. I'm just waiting for this PR to land before I open the next PR.